### PR TITLE
external/esp_idf_port: add ESP32 wifi utility implement.

### PR DESCRIPTION
**Pull request by ChinaTizenRT**
Add ESP32 wifi utility code for intergating with wifi manager.

### CODE REVIEW

File: external/esp_idf_port/esp32/esp32_wifi_utils.c [hunk 1]
```diff
@@ -0,0 +1,740 @@
...
+		break;
+	}
+	case SYSTEM_EVENT_STA_DISCONNECTED: {
+		s_retry_num++;
+		int reason = event->event_info.disconnected.reason;
+		ESP_LOGI(TAG, "[%s] connect to the AP %s fail %d: %d; %d\n", __func__, event->event_info.disconnected.ssid, s_retry_num, event->event_info.disconnected.reason, isStaConnected);
+
+		if (s_retry_num > 0) {
+
+			printf("g_cbk.sta_connected fail\n");
```
Comment (SRCN-LiQin) [inline external/esp_idf_port/esp32/esp32_wifi_utils.c]: 1. blank line,
2. printf in this file, replace with debug log

Comment (ChinaTizenRT) [reply external/esp_idf_port/esp32/esp32_wifi_utils.c]: right, done

File: external/esp_idf_port/esp32/esp32_wifi_utils.c [hunk 2]
```diff
@@ -0,0 +1,737 @@
...
+		,
+	};
+
+	ret = esp_wifi_set_mode(WIFI_MODE_STA);
+	if (ret) {
+		ESP_LOGI(TAG, "esp_wifi_set_mode failed, %d\n", ret);
+		return;
+	}
+
+	ret = esp_wifi_set_config(ESP_IF_WIFI_STA, &wifi_config);
```
Comment (xixidodo) [inline external/esp_idf_port/esp32/esp32_wifi_utils.c:347]: during wifi init,  wifi mode and config is set here.  any special reason here?
it seems that this shall be done in *_start_sta  or *_start_softap.

Comment (ChinaTizenRT) [reply external/esp_idf_port/esp32/esp32_wifi_utils.c:347]: yes, defaulty wifi is enter station mode in some utc and wm_test test.
so in init function, set mode and config.

**Pull request merged**